### ISSUE TITLE
GFA alignments failed

### ISSUE THREAD

**Issue by campenr** [marbl/canu#545]
Hello

I am hitting an issue with canu failing at:

    cd unitigging/4-unitigger
    ./alignGFA.sh
    GFA alignments failed.

I am trying to assemble the lamda sequence reads from the ONP lambda control experiment (1D rapid sequencing kit). I have successfully run canu using the demo *E. coli* data set with no problems, but keep hitting this issue with my lambda 1D sequence reads.

According to metrichor the lambda control experiment worked fine , with 84.8% average accuracy and 97.7% average identity.

I am using canu snapshot v1.5 +128 changes (r8328 84017310563aad430b986f5bae350a49c748fe9d).

I ran the command:
    canu -p lambda -d lambda_1 genomeSize=48500 -nanopore-raw lambda.fasta

I tried running `./alignFGA.sh` directly as that appeared to work for someone else (#485), but that failed also.
I also tried setting `correctedErrorRate` to 0.75 (also other numbers up to 5.0) to see if poor quality overlap could be my problem, but that didn't help either.

Is this an issue with poor quality reads perhaps? I am unsure what files to be looking at to get more info about what went wrong.

Cheers
Richard

**Comment by campenr** [comment]
I looked for the files you mention above, and while some of them are present, neither the *.bed nor the *.bed.gfa files exist. 

Several of the error files have a warning about there being no tigs in the store *.utgStore, and then others complain about the .bed files missing. I can not find the .utgStore file in the output directory either so perhaps this file has not been created hence why it's empty?

Would this indicate that none of my sequences made it to this stage, and it's failing because there are no sequences?

**Comment by brianwalenz** [comment]
Thanks for the reads.

The assembled contig was flagged by a filter that screens out contigs that look to be garbage.  The stereotypical example is a single long low-quality read that had one short read assembled into it:
```
----------------------------
    ----
```
Had the short read not been assembled into it, the long read would have ended up as an unassembled singleton read and discarded.  We added a set of filters to get rid of these contigs.  Unfortunately, the filters also flag read-sized objects (usually plasmids) as unassembled junk.

This will be fixed in the next release.  For now, a workaround is described at http://canu.readthedocs.io/en/latest/faq.html#my-asm-contigs-fasta-is-empty-why.

**Comment by campenr** [comment]
Thanks for taking a look at them for me, I really appreciate it.

I will try with the workaround and see how it goes.

**Comment by campenr** [comment]
Worked like a charm. Thanks again.

**Comment by SarahBM-NU** [comment]
Hello,

I have pretty the same kind of issue when trying to run canu with some bacillus sequences. Most of the sequences are quite short (<999b) as this our first test run with the Minion. I ran canu with the workaround you gave and these parameters: 

 ./canu  -p bacillus -d bacillusdir  genomeSize=5.2m  -nanopore-raw AdamSeq.fastq -minReadLength=500 -contigFilter="2 1000 1.0 1.0 2"

but it ended like this:

BEGIN ASSEMBLY
--
----------------------------------------
-- Starting command on Tue Jul 11 21:41:08 2017 with 360.882 GB free disk space

    cd unitigging/4-unitigger
    ./alignGFA.sh
GFA alignments failed.

-- Finished on Tue Jul 11 21:41:08 2017 (lickety-split) with 360.882 GB free disk space
----------------------------------------
ERROR:
ERROR:  Failed with exit code 1.  (rc=256)
ERROR:

I don't have neither the *.bed nor the *.bed.gfa files but contrarily to Richard I have a bacillus.utgStore folder. If of interest, the content of the partitionedReads.log file within this folder is the following: 
tgStore::tgStore()-- WARNING:  no tigs in store './bacillus.utgStore' version '1'.
For 5 reads, will make 1 partition with up to 6 reads.
Found 5 unpartitioned reads and maximum partition of 0

And just in case, I have attached the .report file I got and the fastq file I used.

[bacillus.report.txt](https://github.com/marbl/canu/files/1140778/bacillus.report.txt)
[AdamSeq.fastq.txt](https://github.com/marbl/canu/files/1140779/AdamSeq.fastq.txt)

Could you please help me figuring out what I did wrong?
Thank you very much in advance for your help!

Sarah

**Comment by campenr** [comment]
Hey Sarah. So I was looking for a .utgStore file and couldn't find one(I didn't realise it was not a file but in fact a folder). I did later find the .utgStore folder, but at that point the problem had been resolved. I had the same warning in a file in the .utgStore folder that you are reporting too.

The 1000 number in the contig filter setting sets the size filter ([relevant page](https://canu.readthedocs.io/en/latest/parameter-reference.html)), but you say your sequences are mostly less than 999bp, so perhaps setting the filter to something like "2 750 0.75 0.75 2" would work better for you?

**Comment by brianwalenz** [comment]
There just aren't enough reads to do anything with.

The 'report' gives a length histogram of the reads used in each of the three stages.  Just before the histogram it'll report how many reads there are:

Your input has:
```
- In gatekeeper store 'correction/test.gkpStore':
--   Found 667 reads.
--   Found 2109622 bases (0.4 times coverage).
```
After read correction:
```
-- In gatekeeper store 'trimming/test.gkpStore':
--   Found 140 reads.
--   Found 381170 bases (0.07 times coverage).
```
After read trimming:
```
-- In gatekeeper store 'unitigging/test.gkpStore':
--   Found 5 reads.
--   Found 4536 bases (0 times coverage).
```
While crashing isn't really the correct behavior here, it really isn't much worse than generating an empty contig.fasta output.  Both cases leave you puzzling over what went wrong.

**Comment by SarahBM-NU** [comment]
Hey campenr, hey Brian,

Thank you very much much for your help! I just wanted to give feedbacks to your suggestion campenr. I tried actually to set the filter at "2 500 0.75 0.75 2" and to "2 500 1.0 1.0 2" as well but unfortunately it ended the same way. 

I think Brian predicted right, we probably need to get more and longer sequences I guess, the fastq file I got this time is only 4.2Mb.

Thanks a lot for your enlightenment though guys, I really appreciate.

Sarah
